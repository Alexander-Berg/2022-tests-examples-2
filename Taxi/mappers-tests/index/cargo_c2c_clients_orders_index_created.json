[
  {
    "expected": [
      {
        "_dummy": null,
        "created_ts": 1586869198.138525,
        "order_id": "text",
        "order_provider_id": "text",
        "phone_pd_id": "text"
      }
    ],
    "input": {
      "created_ts": {
        "$date": "2020-04-14T15:59:58.138525+03"
      },
      "etag": "text",
      "order_id": "text",
      "order_provider_id": "text",
      "phone_pd_id": "text",
      "resolution": "text",
      "roles": [
        "a",
        "b"
      ],
      "terminated_at": {
        "$date": "2020-04-14T15:59:58.138525+03"
      },
      "updated_ts": {
        "$date": "2020-04-14T15:59:58.138525+03"
      },
      "user_id": "text"
    }
  }
]
