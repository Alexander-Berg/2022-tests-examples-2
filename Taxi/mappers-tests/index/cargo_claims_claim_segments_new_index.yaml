  - input:
        claims_created_ts: "2020-07-29T16:48:33.030330+03:00"
        claims_updated_ts: "2020-07-30T00:16:06.494489+03:00"
        raw:
          - allow_batch: false
            cargo_order_id: "66c93b5-9d6f-4f7a-bb3f-cff61572a6d3"
            claim_id: 209149
            claim_uuid: "824ea04ca0dd419f9300a425bd4fbfe9"
            created_ts: "2020-07-29T19:48:41.817265+03:00"
            dispatch_revision: 4
            id: 1
            paper_act: false
            perform_order: 1
            points_user_version: 1
            provider_order_id: "09221642b7f42a388f9f87532952bfa9"
            resolution: "failed"
            revision: 4
            special_requirements:
                virtual_tariffs: []
            status: "failed"
            updated_ts: "2020-07-30T03:16:06.494489+03:00"
            uuid: "f08ffb59-cc5d-4271-8617-a05b5b3371c9"
        replication_id: "824ea04ca0dd419f9300a425bd4fbfe9"
    expected:
      - segment_uuid: "f08ffb59-cc5d-4271-8617-a05b5b3371c9"
        claim_uuid: "824ea04ca0dd419f9300a425bd4fbfe9"
        _dummy:
