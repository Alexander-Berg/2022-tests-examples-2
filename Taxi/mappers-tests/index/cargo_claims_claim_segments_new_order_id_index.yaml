  - input:
        claims_created_ts:
            $date: "2020-07-29T16:48:33.030330+03:00"
        claims_updated_ts:
            $date: "2020-07-30T00:16:06.494489+03:00"
        raw:
          - claim_uuid: "0000c26402ee4074ae40042256db25bd"
            cargo_order_id: "429b4f25-e9d0-41b2-9fe8-906257133f39"
          - claim_uuid: "0000c26402ee4074ae40042256db25bd"
            cargo_order_id: "429b4f25-e9d0-41b2-9fe8-906257133f38"
          - claim_uuid: "0000c26402ee4074ae40042256db25bd"
        replication_id: "0000c26402ee4074ae40042256db25bd"
    expected:
      - cargo_order_id: "429b4f25-e9d0-41b2-9fe8-906257133f39"
        claim_uuid: "0000c26402ee4074ae40042256db25bd"
        _dummy:
      - cargo_order_id: "429b4f25-e9d0-41b2-9fe8-906257133f38"
        claim_uuid: "0000c26402ee4074ae40042256db25bd"
        _dummy:
