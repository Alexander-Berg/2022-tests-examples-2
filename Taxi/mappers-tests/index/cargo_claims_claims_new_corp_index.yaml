  - input:
        replication_id: "000017a1a1174cc7945fbd6dc0bf856d"
        corp_client_id: "7ff7900803534212a3a66f4d0e114fc2"
        uuid_id: "000017a1a1174cc7945fbd6dc0bf856d"
        created_ts:
            $date: "2020-02-11T11:55:10.894238"
        claims_created_ts:
            $date: "2020-02-11T11:55:10.894238"
        claims_updated_ts:
            $date: "2020-08-04T14:16:25.778932"
    expected:
      - corp_client_id: "7ff7900803534212a3a66f4d0e114fc2"
        claim_created_ts: -1581422110
        claim_uuid: "000017a1a1174cc7945fbd6dc0bf856d"
        _dummy:

  - input:
        replication_id: "000017a1a1174cc7945fbd6dc0bf85xx"
        uuid_id: "000017a1a1174cc7945fbd6dc0bf85xx"
        created_ts:
            $date: "2020-02-11T11:55:10.894238"
        claims_created_ts:
            $date: "2020-02-11T11:55:10.894238"
        claims_updated_ts:
            $date: "2020-08-04T14:16:25.778932"
    expected: []
