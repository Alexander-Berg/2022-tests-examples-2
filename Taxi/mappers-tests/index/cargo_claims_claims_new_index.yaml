  - input:
        uuid_id: 987cfe7d6ab142ee811ab8e5f647a5fe
        claims_created_ts:
            $a:
                raw_type: datetime
            $v: '2020-02-11T11:55:10.894238+00:00'
    expected:
      - claims_created_ts: -1581422110
        claim_uuid: 987cfe7d6ab142ee811ab8e5f647a5fe
        _dummy:
