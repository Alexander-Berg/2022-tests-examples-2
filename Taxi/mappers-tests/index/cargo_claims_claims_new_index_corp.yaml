  - input:
        corp_client_id: 7ff7900803534212a3a66f4d0e114fc2
        claims_created_ts:
            $a:
                raw_type: datetime
            $v: '2020-02-11T11:55:10.894238+00:00'
        uuid_id: 987cfe7d6ab142ee811ab8e5f647a5fe
    expected:
      - corp_client_id: 7ff7900803534212a3a66f4d0e114fc2
        claims_created_ts: -1581422110
        claim_uuid: 987cfe7d6ab142ee811ab8e5f647a5fe
        _dummy:

  - input:
        claims_created_ts:
            $a:
                raw_type: datetime
            $v: '2020-02-11T11:55:10.894238+00:00'
        uuid_id: 987cfe7d6ab142ee811ab8e5f647a5fe
    expected: []
