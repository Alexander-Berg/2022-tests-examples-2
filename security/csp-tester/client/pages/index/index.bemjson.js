module.exports = {
    block: 'page',
    title: 'CSP tester',
    favicon: '/favicon.ico',
    head: [
        { elem: 'meta', attrs: { name: 'description', content: 'Test your CSP for errors online' } },
        { elem: 'meta', attrs: { name: 'viewport', content: 'width=device-width, initial-scale=1' } },
        { elem: 'css', url: 'index.min.css' }
    ],
    scripts: [{ elem: 'js', url: 'index.min.js', i18n: true }],
    mods: { theme: 'islands' },
    content: [
        {
            block: 'ng-head',
            service: {
                id: 'csp',
                name: 'CSP tester'
            },
            left: {
                block: 'ng-headmenu',
                content: [
                    {
                        elem: 'item',
                        url: 'https://wiki.yandex-team.ru/product-security/csp/',
                        text: 'Узнать больше о внедрении CSP и рекомендациях по составлению политики'
                    }
                ]
            },
            right: [
                // {
                //     block: 'ng-head-user',
                //     uid: 1,
                //     login: 'tadatuta',
                //     avatar: '//center.yandex-team.ru/api/v1/user/tadatuta/avatar/50.jpg'
                // }
            ]
        },
        {
            block: 'main',
            content: [
                {
                    block: 'form',
                    method: 'post',
                    mix: { block: 'test-form', js: true },
                    content: [
                        {
                            elem: 'control',
                            content: [
                                {
                                    block: 'textarea',
                                    mods: { theme: 'islands', size: 'm', width: 'available' },
                                    name: 'policy',
                                    placeholder: 'Введите сюда вашу CSP политику или URL для её получения через заголовок'
                                }
                            ]
                        },
                        {
                            block: 'button',
                            mods: { theme: 'islands', size: 'm', type: 'submit', view: 'action', disabled: true },
                            text: 'Проверить'
                        }
                    ]
                },
                {
                    block: 'result',
                    content: [
                        {
                            elem: 'valid',
                            content: {
                                block: 'heading',
                                mods: { level: 1 },
                                content: 'Ошибок не обнаружено'
                            }
                        },
                        {
                            elem: 'invalid'
                        },
                        {
                            elem: 'content'
                        }
                    ]
                }
            ]
        },
        {
            block: 'ng-foot',
            content: [
                {
                    block: 'link',
                    mods: { theme: 'islands' },
                    mix: { block: 'ng-foot', elem: 'nda' },
                    url: 'https://wiki.yandex-team.ru/product-security/csp/tester/#feedback',
                    content: 'Обратная связь'
                },
                {
                    block: 'link',
                    mods: { theme: 'islands' },
                    mix: { block: 'ng-foot', elem: 'nda' },
                    url: 'https://wiki.yandex-team.ru/hr/nda/',
                    content: 'Конфиденциально'
                },
                {
                    elem: 'copyright',
                    content: '© ' + new Date().getFullYear() + ' ООО «Яндекс»'
                }
            ]
        }
    ]
};
