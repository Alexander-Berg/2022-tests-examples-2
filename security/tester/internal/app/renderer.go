package app

import (
	"bytes"
	_ "embed"
	"fmt"
	"io"
	"strings"
	"time"

	"golang.org/x/crypto/ssh"
	"golang.org/x/crypto/ssh/agent"

	"a.yandex-team.ru/security/skotty/skotty/pkg/sshutil"
)

func (a *App) renderClientInfo(cl *ClientInfo) ([]byte, error) {
	formatBool := func(b bool) string {
		if b {
			return "yes"
		}
		return "no"
	}

	staffedKey := func(key ssh.PublicKey) string {
		if cl.StaffkeysErr != nil {
			return fmt.Sprintf("unble to check: %s", cl.StaffkeysErr)
		}

		var target string
		if cert, ok := key.(*ssh.Certificate); ok {
			target = ssh.FingerprintSHA256(cert.SignatureKey)
		} else {
			target = ssh.FingerprintSHA256(key)
		}

		for _, sk := range cl.StaffKeys {
			if sk == target {
				return formatBool(true)
			}
		}

		return formatBool(false)
	}

	formatKey := func(out io.Writer, in ssh.PublicKey) {
		writeCertInfo := func(key ssh.PublicKey) {
			crt, ok := key.(*ssh.Certificate)
			if !ok {
				return
			}

			_, _ = fmt.Fprintf(out, "      - revoked: %s\n", formatBool(a.krl.IsRevoked(crt)))
			_, _ = fmt.Fprintf(out, "      - principals: %s\n", strings.Join(crt.ValidPrincipals, ","))
			_, _ = fmt.Fprintf(out, "      - valid before: %s\n", time.Unix(int64(crt.ValidBefore), 0).Format("2006-01-02T15:04"))
			_, _ = fmt.Fprintf(out, "      - ca fingerprint: %s\n", sshutil.Fingerprint(crt.SignatureKey))
			_, _ = fmt.Fprintf(out, "      - serial: %d\n", crt.Serial)
			_, _ = fmt.Fprintf(out, "      - key id: %s\n", crt.KeyId)
		}

		writeKeyInfo := func(key ssh.PublicKey) {
			_, _ = fmt.Fprintf(out, "    * %s\n", sshutil.Fingerprint(key))
			_, _ = fmt.Fprintf(out, "      - type: %s\n", key.Type())
			_, _ = fmt.Fprintf(out, "      - present on staff: %s\n", staffedKey(key))
			writeCertInfo(key)
		}

		switch v := in.(type) {
		case *agent.Key:
			sshKey, err := ssh.ParsePublicKey(v.Marshal())
			if err != nil {
				break
			}

			writeKeyInfo(sshKey)
			_, _ = fmt.Fprintf(out, "      - comment: %s\n", v.Comment)
		default:
			writeKeyInfo(v)
		}
	}

	var out bytes.Buffer
	_, _ = fmt.Fprint(&out, "Welcome to the Skotty Validation server!\n\n")

	if cl.ClientCompatNotice != "" {
		_, _ = fmt.Fprintln(&out, cl.ClientCompatNotice)
	}
	_, _ = fmt.Fprintln(&out, "Info about your connection:")
	_, _ = fmt.Fprintf(&out, "  - Client version: %s\n", cl.ClientVersion)
	_, _ = fmt.Fprintf(&out, "  - Username: %s\n", cl.Username)
	_, _ = fmt.Fprintf(&out, "  - Auth keys: %d\n", len(cl.AuthKeys))
	for _, key := range cl.AuthKeys {
		formatKey(&out, key)
	}

	if cl.AgentForwarded {
		_, _ = fmt.Fprintln(&out, "  - Agent forwarded: yes")
		_, _ = fmt.Fprintf(&out, "  - Agent support sudo: %s\n", formatBool(cl.SudoAvailable))
		_, _ = fmt.Fprintf(&out, "  - Agent keys: %d\n", len(cl.AgentKeys))
		for _, key := range cl.AgentKeys {
			formatKey(&out, key)
		}
	} else {
		_, _ = fmt.Fprintln(&out, "  - Agent forwarded: no")
	}

	_ = out.WriteByte('\n')
	_, _ = out.WriteString("Have a great day! ;)\n")
	return out.Bytes(), nil
}
