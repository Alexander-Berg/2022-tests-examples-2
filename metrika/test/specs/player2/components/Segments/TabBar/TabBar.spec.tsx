import React from 'react'
import renderer from 'react-test-renderer'
import { TabBar } from '@player2/components/UI/TabBar/index'
import 'jest-styled-components'

describe('<TabBar>', () => {
  it('Renders correctly (shows with 2 or more tabs)', () => {
    let props: any = {
      tabs: [],
    }
    let tree = renderer
      .create(<TabBar {...props}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
    props = {
      tabs: [
        {
          tabId: '1',
          active: true,
          title: 'title1',
        },
      ],
    }
    tree = renderer
      .create(<TabBar {...props}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
    props = {
      tabs: [
        {
          tabId: '1',
          title: 'title1',
          active: true,
        },
        {
          tabId: '2',
          title: 'title2',
          active: false,
        },
      ],
    }
    tree = renderer
      .create(<TabBar {...props}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
