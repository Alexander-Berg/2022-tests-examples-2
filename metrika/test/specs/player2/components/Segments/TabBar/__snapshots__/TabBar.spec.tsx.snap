// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBar> Renders correctly (shows with 2 or more tabs) 1`] = `
.c1 {
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-transition: height 100ms ease;
  transition: height 100ms ease;
  height: 0px;
}

<div
  className="c0 c1"
/>
`;

exports[`<TabBar> Renders correctly (shows with 2 or more tabs) 2`] = `
.c1 {
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #000;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 250px;
  border-radius: 3px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 10px;
  background-color: #fff;
  box-sizing: border-box;
  opacity: 1;
}

.c2:not(:first-child) {
  margin-left: 2px;
}

.c3 {
  height: 30px;
  width: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 20px;
  font-weight: bold;
  padding-left: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c4 {
  height: 30px;
  width: 16px;
  margin: 0 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 16px;
  height: 16px;
  background-color: #fc0;
}

.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-transition: height 100ms ease;
  transition: height 100ms ease;
  height: 0px;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
    style={
      Object {
        "maxWidth": "calc((100% - 14px) / 1)",
      }
    }
  >
    <div
      className="c3"
    >
      A
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
      />
    </div>
    <div
      className="c6"
    >
      title1
    </div>
  </div>
</div>
`;

exports[`<TabBar> Renders correctly (shows with 2 or more tabs) 3`] = `
.c1 {
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #000;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 250px;
  border-radius: 3px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 10px;
  background-color: #fff;
  box-sizing: border-box;
  opacity: 1;
}

.c2:not(:first-child) {
  margin-left: 2px;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #000;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 250px;
  border-radius: 3px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 10px;
  background-color: #fff;
  box-sizing: border-box;
  opacity: 0.4;
}

.c7:not(:first-child) {
  margin-left: 2px;
}

.c3 {
  height: 30px;
  width: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 20px;
  font-weight: bold;
  padding-left: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c4 {
  height: 30px;
  width: 16px;
  margin: 0 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 16px;
  height: 16px;
  background-color: #fc0;
}

.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-transition: height 100ms ease;
  transition: height 100ms ease;
  height: 40px;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
    style={
      Object {
        "maxWidth": "calc((100% - 14px) / 2)",
      }
    }
  >
    <div
      className="c3"
    >
      A
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
      />
    </div>
    <div
      className="c6"
    >
      title1
    </div>
  </div>
  <div
    className="c7"
    style={
      Object {
        "maxWidth": "calc((100% - 14px) / 2)",
      }
    }
  >
    <div
      className="c3"
    >
      A
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
      />
    </div>
    <div
      className="c6"
    >
      title2
    </div>
  </div>
</div>
`;

exports[`<TabBar> Renders correctly (shows with 2 or more tabs) 1`] = `
.c1 {
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-transition: height 100ms ease;
  transition: height 100ms ease;
  height: 0px;
}

<div
  className="c0 c1"
/>
`;

exports[`<TabBar> Renders correctly (shows with 2 or more tabs) 2`] = `
.c1 {
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #000;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 250px;
  border-radius: 3px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 10px;
  background-color: #fff;
  box-sizing: border-box;
  opacity: 1;
}

.c2:not(:first-child) {
  margin-left: 2px;
}

.c3 {
  height: 30px;
  width: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 20px;
  font-weight: bold;
  padding-left: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c4 {
  height: 30px;
  width: 16px;
  margin: 0 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 16px;
  height: 16px;
  background-color: #fc0;
}

.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-transition: height 100ms ease;
  transition: height 100ms ease;
  height: 0px;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
    style={
      Object {
        "maxWidth": "calc((100% - 14px) / 1)",
      }
    }
  >
    <div
      className="c3"
    >
      A
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
      />
    </div>
    <div
      className="c6"
    >
      title1
    </div>
  </div>
</div>
`;

exports[`<TabBar> Renders correctly (shows with 2 or more tabs) 3`] = `
.c1 {
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #000;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 250px;
  border-radius: 3px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 10px;
  background-color: #fff;
  box-sizing: border-box;
  opacity: 1;
}

.c2:not(:first-child) {
  margin-left: 2px;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #000;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 250px;
  border-radius: 3px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 10px;
  background-color: #fff;
  box-sizing: border-box;
  opacity: 0.4;
}

.c7:not(:first-child) {
  margin-left: 2px;
}

.c3 {
  height: 30px;
  width: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 20px;
  font-weight: bold;
  padding-left: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c4 {
  height: 30px;
  width: 16px;
  margin: 0 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 16px;
  height: 16px;
  background-color: #fc0;
}

.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-transition: height 100ms ease;
  transition: height 100ms ease;
  height: 40px;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
    style={
      Object {
        "maxWidth": "calc((100% - 14px) / 2)",
      }
    }
  >
    <div
      className="c3"
    >
      A
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
      />
    </div>
    <div
      className="c6"
    >
      title1
    </div>
  </div>
  <div
    className="c7"
    style={
      Object {
        "maxWidth": "calc((100% - 14px) / 2)",
      }
    }
  >
    <div
      className="c3"
    >
      A
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
      />
    </div>
    <div
      className="c6"
    >
      title2
    </div>
  </div>
</div>
`;
